src/view/gui/AppearancePage.cpp

```diff
@@ -54,6 +57,9 @@ AppearancePage::~AppearancePage()
 
 void AppearancePage::OnChangeTheme(int theme)
 {
+    // OnChangeTheme updates the instance variables to contain the selected theme and style sheet but does not apply it.
+    m_PreviousTheme = m_SelectedTheme;
```

Comment:
These 2 lines may not be needed here. We update the previous with selected only when we hit okay.